### ISSUE TITLE
Result replication

### ISSUE THREAD

**Issue by mahendrathapa** [HarliWu/FIARSE#1]
Thank you, @HarliWu, for releasing the code for this interesting work. However, I am unable to replicate the results. 

Results reported in the paper

![Image](https://github.com/user-attachments/assets/021dfdca-a0a1-4018-970f-7d508da15424)

To replicate results for Cifar10 and Cifar100:
```python -u start.py --root ~/dataset --model BetaResNet18_sbn --dataset cifar10 --partitioner balanced --alpha 0.3 --bsz 20 --num-workers 100 --num-part 10 --lr -1 --gpu-idx 0 --K 100 --T 800 --model-size 1.0 0.25 0.0625 0.015625 --model-dist 25 25 25 25```

```python -u start.py --root ~/dataset --model BetaResNet18_sbn --dataset cifar100 --partitioner balanced --alpha 0.3 --bsz 20 --num-workers 100 --num-part 10 --lr -1 --gpu-idx 0 --K 100 --T 800 --model-size 1.0 0.25 0.0625 0.015625 --model-dist 25 25 25 25```

The result that I got:
Cifar10
![Image](https://github.com/user-attachments/assets/d5fe4437-6281-4c36-92a7-2b41388ea0eb)

Cifar100
![Image](https://github.com/user-attachments/assets/0e61659e-bbe1-423b-b2aa-015df56839e1)

@HarliWu, could you please share the information and configuration needed to replicate the reported results?

**Comment by HarliWu** [comment]
Hi @mahendrathapa, 

Thank you for bringing this up. There are a few factors that might explain the discrepancies in replication:

1. **Model initialization:** Although we set a random seed, we observed that the model parameters were still initialized randomly in some environments. To ensure consistency, we used a fixed random model initialization for our reported experiments.
2. **Choice of hyperparameters:** As mentioned in the paper, we explored various hyperparameter settings, including learning rates from the set ${10^{-1}, 10^{-1.5}, 10^{-2}}$ and momentum values from ${0, 0.9}$. We recommend trying these combinations, as performance can be sensitive to these choices.
3. **Result Reporting:** The results shown in the paper are not strictly from the 800th communication round. Instead, they are selected from a range between the 700th and 900th rounds to account for training fluctuations. We found that reporting results from a single round can be misleading due to potential instability and cannot demonstrate the effectiveness of our proposed FIARSE. 

Please let us know if the issue persists or if you'd like further clarification.

**Comment by mahendrathapa** [comment]
Hi @HarliWu, thank you for your response. 

1.   **Model initialization**: What are your random seed values for cifar10 and cifar100 experiments? The random seeds are set in start.py as 

![Image](https://github.com/user-attachments/assets/645b05e3-666e-49b8-afd3-6cd762f88bd6). 
Do we have to do a random seed setup in any other places?

2. **Choice of hyperparameters**: It would be helpful if you could share the hyperparameters used for the results reported in Table 1.

3.  **Result Reporting**: What is your mechanism for choosing a model for reporting results? Is it based on train loss for sampled clients, train accuracy for sampled clients, test loss for all clients, or test accuracy for all clients? 

It would be quite helpful if you could share the git branch from which we can replicate the Table 1 results.

**Comment by HarliWu** [comment]
Hi, 

I'm sorry for not getting back to you sooner due to my busy schedule recently. I believe the default configuration can provide the best result, and I try to figure out why your running results look so different from ours. Here are my training logs for CIFAR-100: 

Results at Round 760:

<img width="2355" height="427" alt="Image" src="https://github.com/user-attachments/assets/c45ec0be-cbfa-4109-a09b-27311b4da9f6" />

Results at Round 800:

<img width="2364" height="424" alt="Image" src="https://github.com/user-attachments/assets/0aaedc01-59e6-4ec8-abd9-f1a0a7fb322b" />

Results at Round 850:

<img width="2345" height="425" alt="Image" src="https://github.com/user-attachments/assets/ac7b54e2-1c1f-457d-a69e-aeba34448d27" />

I also discussed with a co-author how the results were collected. He shared the following table, which records the best performance of models of different sizes within the range of 750 to 850 training rounds. For the "average" column, we computed the average of each model's best performance, rather than reporting the performance of a single model. The same procedure was applied to the other baseline approaches as well. 

<img width="1743" height="253" alt="Image" src="https://github.com/user-attachments/assets/0558b0a7-b154-4447-9d52-f6088aaa2456" />
